src/synthesiser/Synthesiser.cpp

```diff
@@ -15,14 +15,14 @@
  ***********************************************************************/
 
 #include "synthesiser/Synthesiser.h"
-#include "AggregateOp.h"
-#include "BinaryConstraintOps.h"
-#include "FunctorOps.h"
 #include "Global.h"
 #include "RamTypes.h"
-#include "RelationTag.h"
 #include "SymbolTable.h"
 #include "json11.h"
+#include "parser/AggregateOp.h"
+#include "parser/BinaryConstraintOps.h"
+#include "parser/FunctorOps.h"
+#include "parser/RelationTag.h"
```

Comment:
These are not parser files. Their use in the Synthesiser, Interpreter, RAM, and to a lesser extent AST, are examples of this. They are closer to datastructure files than parser. The parser should only deal with input datalog and AST.

We should either have separate files for the different components, or some sort of shared directory (currently that's `src`).

Reply:
I thought about this. The point is that they are language objects that go across RAM/AST. I would suggest to see the parser folder not just as a "parser" folder but as a language folder. Developers who wish to extend the language would need to change these objects there. 

I don't like to place files into too many different sub-directories.

Reply:
That sounds pretty reasonable, but what about a `language` folder instead? `parser` can be specifically for parsing datalog, `language` is for the souffle language. It *would* be another folder, but the root src directory is pretty empty after this change, and moving of datastructures.

In large part I object to `parser` for these as they can be public header files, which is quite divorced from parsing. I do really like the idea of moving them out of `src/`, though, so that the purpose of the files is clear from their location.